src/OpenTK.Windowing.Desktop/NativeWindow.cs

```diff
@@ -48,13 +48,10 @@ public class NativeWindow : INativeWindow
         /// <inheritdoc />
         public KeyboardState LastKeyboardState { get; private set; }
 
-        private readonly JoystickState[] _joystickStates = new JoystickState[16];
+        private readonly IJoystickState[] _joystickStates = new IJoystickState[16];
 
         /// <inheritdoc/>
-        public JoystickState[] JoystickStates { get => _joystickStates; }
-
-        /// <inheritdoc/>
-        public JoystickState[] LastJoystickStates { get; private set; }
+        public IReadOnlyCollection<IJoystickState> JoystickStates { get => _joystickStates; }
```

Comment:
`IReadOnlyCollection<T>` is not a great interface. It sounds like it would support the read-only subset of the `ICollection<T>` interface but it doesn't. It doesn't even have an indexer property. If this is the only interface that will do and we have to settle for it, it's fine I guess.
An alternative would be to introduce functions to get individual `JoystickStates` but not sure that would be any better.

Reply:
I just found that single dimension arrays also `IReadOnlyList<T>` which is a much nicer interface with an indexer.

Reply:
I'd prefer not to go down the IReadOnlyList route, because that involves creating a new List<IJoystickState> each time the property getter is called.

We could create an ArrayAccessor<T> class with an indexer with public get and private set, which would just wrap the array - then declare _joystickStates as an instance of that class. When we start to talk about it, it seems like one of those things Microsoft should already have thought of.

Reply:
OK, I'm going to create it as List<IJoystickState> and return it as IReadOnlyList<IJoystickState>